### PR TITLE
Dynamic widget documentation

### PR DESCRIPTION
**Pull request by matthewlilley**
- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines]
##### Affected core subsystem(s)
doc
##### Refers/Fixes
https://github.com/gitcoinco/web/issues/354

### CODE REVIEW

**Comment by matthewlilley** [comment]
@mbeacom Moved this to a separate branch on my end.

**Comment by mbeacom** [comment]
@matthewlilley Can you PR back the contents of your SDK repo *into* this repository under it's own subdirectory?

**Comment by matthewlilley** [comment]
@mbeacom Are able to copy from node_modules/gitcoin-sdk in your build pipeline? I think it makes sense for the repository to be isolated since it's available as an npm module as well. I think it would be best to transfer the repository over to your organisation, and the same with the npm listing.d

**Comment by matthewlilley** [comment]
@mbeacom What do you think? The concern here is having 2 different sources of truth.

Consider the need to update the NPM module, but the web version is not updated because it would need to be copied into this repo manually, instead of just updating the version number in your package json. 

If you have an idea of another way this would work, I'm all ears.

**Comment by thelostone-mc** [comment]
@matthewlilley you've got a valid point there -> but what we could do it is let the sdk reside in it's own subdirectory & any time an update has to be made, we publish it to npm from this subdirectory ?

It's just that at our current state -> figured it would be easier to maintain if we could it have it one place. Your thoughts on that ?

**Comment by owocki** [comment]
hey guys -- just checking in here.. we have a client who wants to use the widget for a hackathon in a week and a half.  

i'm down with the proposal that @thelostone-mc left on the comment above, as long as we document it..

**Comment by owocki** [comment]
Question: is CORS gonna be a problem for this?  I get [this error](http://bits.owocki.com/020j302T2z31/Screen%20Shot%202018-05-08%20at%204.08.27%20PM.png) when I try to test the js code locally

**Comment by matthewlilley** [comment]
@thelostone-mc Sounds good. Okay, I've added the source into a "js-sdk" sub-directory at root level.
@owocki  I used a cors proxy to get around that in development, but yes, if this endpoint is going to be used publicly, it should allow for requests from origins other than it's own.

**Comment by owocki** [comment]
@mbeacom is there an easy way to get around the CORS stuff for the API?  i googled around best practices for doign this with DRF but didnt figure it out before something else came up

**Comment by matthewlilley** [comment]
@owocki I know that question wasn't directed at me, but the simple answer is ALLOW_ALL for unrestricted endpoints, in this case, the "VIEW (GET)" of the bounties resource.

File: js-sdk/package.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "build": "webpack --config webpack.config.js --mode production --progress --colors",
+    "prepublish": "npm run build",
+    "eslint": "./node_modules/eslint/bin/eslint.js --quiet --config .eslintrc ./src/*.js",
+    "eslint:fix": "./node_modules/eslint/bin/eslint.js --quiet --fix --config .eslintrc ./src/*.js",
+    "stylelint": "stylelint ./src/*.css",
+    "stylelint:fix": "stylelint --fix ./src/*.css",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "keywords": ["Gitcoin", "SDK"],
+  "author": "Matthew Lilley",
```
Comment (thelostone-mc) [inline js-sdk/package.json]: @owocki  Do we want this updated ?

**Comment by matthewlilley** [comment]
@owocki @mbeacom Allow all for GET requests of bounties resource. https://github.com/gitcoinco/web/pull/1126

**Comment by owocki** [comment]
hott.. testing now

File: js-sdk/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+Importing the SDK into your application will attempt to autoload the widget by searching for '.gitcoin-widget' selectors
+
+```javascript
+import 'gitcoin-sdk';
+```
+or
+```javascript
+require('gitcoin-sdk');
+```
+
```
Comment (owocki) [inline js-sdk/README.md]: should we include a webified version in the documentation?

**Comment by owocki** [comment]
i'd love to see a "webified" example in the documentation.
between that and the CORS PR, this looks like it's getting close.

File: js-sdk/package.json (hunk 1 above)
Comment (owocki) [reply js-sdk/package.json]: where is the info displayed? on npm somewhere?

File: js-sdk/README.md (hunk 2 above)
Comment (matthewlilley) [reply js-sdk/README.md]: Hello @owocki,

The library should be targeted for the web, the simplest way to use it as a webmasteris by following the initial Example, (Part1 & Part2), this follows the same method used by Facebook and Twitter for inclusion of their own widgets. 

For example: https://developers.facebook.com/docs/plugins/like-button

File: js-sdk/package.json (hunk 1 above)
Comment (matthewlilley) [reply js-sdk/package.json]: https://www.npmjs.com/package/gitcoin-sdk

You can edit this and republish if you wish. I'll need somebody to take control of the NPM package on npmjs. 

Do you have an npmjs org account for Gitcoin? 

If not, go ahead and create one https://www.npmjs.com/org/create, once done I'll transfer the package over to it.

File: js-sdk/README.md (hunk 2 above)
Comment (owocki) [reply js-sdk/README.md]: ah you are right.. thanks

**Comment by owocki** [comment]
@matthewlilley i like that idea.. want to create a v2 ticket scoping it out?  ill probably bounty it once its created

**Comment by owocki** [comment]
it works! yay http://bits.owocki.com/2E2q2Y3o0R1X/foo.html

File: js-sdk/README.md (hunk 2 above)
Comment (matthewlilley) [reply js-sdk/README.md]: > ah you are right.. thanks

The documentation isn't clear enough about that, I'll make some updates to it shortly.

File: js-sdk/package.json (hunk 1 above)
Comment (matthewlilley) [reply js-sdk/package.json]: I think I need to transfer it to a user which has access to the org. Can you confirm your NPM username?

Comment (mbeacom) [reply js-sdk/package.json]: @matthewlilley `mbeacom` 😂

Comment (matthewlilley) [reply js-sdk/package.json]: @mbeacom I had to make sure 😂 Done!

**Comment by matthewlilley** [comment]
@owocki  

> @matthewlilley i like that idea.. want to create a v2 ticket scoping it out? ill probably bounty it once its created

Sure!

**Pull request merged**